VisualStudio.gitignore

```diff
@@ -22,24 +24,37 @@ bld/
 [Bb]in/
 [Oo]bj/
 
-# Visual Studio 2015 cache/options directory
-.vs/
-
-# MSTest test Results
+# MSTest
+# a test framework from Microsoft
+# remove: if you are not using the MSTest runner to run your tests
 [Tt]est[Rr]esult*/
 [Bb]uild[Ll]og.*
 
-# NUNIT
+# NUnit
+# an open source test framework for .NET
+# site: http://www.nunit.org/
+# remove: if you are not using the NUnit test runner to run your tests
 *.VisualState.xml
 TestResult.xml
 
-# Build Results of an ATL Project
+# ATL (Active Template Library) Project
+# remove: if you do not have an ATL project in your solution
 [Dd]ebugPS/
 [Rr]eleasePS/
 dlldata.c
 
 # DNX
+# summary: the environment for cross-platform .NET development
+# site: https://github.com/aspnet/dnx
+#
+# locks the package references for a project to specific versions
+# url: https://github.com/aspnet/Home/wiki/Lock-file
+# remove: if you would like to lock your dependencies to a specific version
+#         when committing and prevent package restore from selecting a
+#         more recent version
 project.lock.json
```

Comment:
This change was submitted in #1505 but a [twitter discussion](https://twitter.com/adamralph/status/616854162735779840) yesterday about the change left things unclear. So here's my attempt at enriching the rule rather than continuing the saga of enabling or disabling this rule.

I've provided :

 - some comments (line 50) about what this file represents
 - a `url` to more information
 - a `remove` rule, which explains the situation when you should be disabling this rule